File: scripts/audit_category_am.py
import os
import re

directories = ["src/pages", "src/components"]

for d in directories:
    for root, _, files in os.walk(d):
        if 'components/ui' in root: continue
        for file in files:
            if not file.endswith('.tsx') and not file.endswith('.ts'): continue
            
            filepath = os.path.join(root, file)
            with open(filepath, 'r') as f:
                content = f.read()

            is_dialog = 'DialogContent' in content or 'SheetContent' in content
            if not is_dialog: continue
            
            # Print any block matching the div flex justify-end pattern
            pattern = re.compile(r'<div\s+className="([^"]*(?:justify-end)[^"]*)"[^>]*>.*?</div>', re.DOTALL)
            for match in pattern.finditer(content):
                block = match.group(0)
                if ('<Button' in block or '<button' in block) and block.count('<div') < 3 and len(block) < 1000:
                    print(f"File: {filepath}")
                    print(block)
                    print("-" * 40)

File: scripts/audit_category_am.ts
import { Project, SyntaxKind } from 'ts-morph';

const project = new Project();
project.addSourceFilesAtPaths(['src/pages/**/*.tsx', 'src/components/**/*.tsx']);

let filesWithMissingFooters = 0;

for (const sourceFile of project.getSourceFiles()) {
  if (sourceFile.getFilePath().includes('components/ui/')) continue;
  
  const text = sourceFile.getFullText();
  const isDialog = text.includes('DialogContent') || text.includes('SheetContent');
  if (!isDialog) continue;
  
  // Does it already have a DialogFooter / SheetFooter?
  const hasFooter = text.includes('DialogFooter') || text.includes('SheetFooter');
  
  if (!hasFooter) {
     // Check if there are buttons in the dialog/sheet at all
     // We can just dump any file that has DialogContent but no DialogFooter, and has Buttons
     const hasButtons = text.includes('<Button') || text.includes('<button');
     if (hasButtons) {
         console.log(`Missing Footer: ${sourceFile.getBaseName()}`);
         filesWithMissingFooters++;
     }
  }
}

console.log(`Files with Dialog/Sheet lacking Footers: ${filesWithMissingFooters}`);
